src/structs/jpeg_write.rs

```diff
@@ -357,59 +357,73 @@ fn encode_block_seq(
     huffw: &mut BitWriter,
     dctbl: &HuffCodes,
     actbl: &HuffCodes,
-    block: &[i16; 64],
+    block: &AlignedBlock,
 ) {
     // encode DC
-    write_coef(huffw, block[0], 0, dctbl);
-
+    write_coef(huffw, block.get_coefficient(0), 0, dctbl);
+
+    // using SIMD instructions, construct a 64 bit mask of all
+    // the non-zero coefficients in the block. This can be used
+    // to efficiently skip zero blocks using trailing zero scan.
+    let b: &[i16x16; 4] = cast_ref(block.get_block());
+    let mut mask = (b[0].cmp_eq(i16x16::ZERO).move_mask() as u64)
+        | ((b[1].cmp_eq(i16x16::ZERO).move_mask() as u64) << 16)
+        | ((b[2].cmp_eq(i16x16::ZERO).move_mask() as u64) << 32)
+        | ((b[3].cmp_eq(i16x16::ZERO).move_mask() as u64) << 48);
```

Comment:
Hmm, we can make this mask even earlier, during Lepton coefficient decoding step. And it will be almost free lunch.

Reply:
Unfortunately it's done a lot later in the process so it would have to be stored somewhere intermediate...